meson.build

```diff
@@ -2221,7 +2224,7 @@ if conf.get('HAVE_PAM') == 1
                 install_dir : rootlibexecdir)
 endif
 
-if conf.get('ENABLE_EFI') == 1 and conf.get('HAVE_BLKID') == 1
+if conf.get('HAVE_BLKID') == 1 and conf.get('HAVE_GNU_EFI') == 1
```

Comment:
So hmm, I know i suggested that if sd-boot is turned off this should also turn off bootctl. But thinking about this, maybe not. The thing is that at least `bootctl reboot-to-firmware` and `bootctl systemd-efi-options` are truly useful outside of sd-boot systems. And the boot loader spec listing logic kinda too, for boot loaders that implement that (fedora's grub claims to).

So I think the conditionalization should be tweaked here: for bootctl itself stick to ENABLE_EFI, but for systemd-bless-boot and its generator use HAVE_GNU_EFI. Because the latter two are really strictly only sd-boot-specific stuff.

Reply:
But we do have `--firmware-setup` option for `systemctl reboot`. As for `bootctl systemd-efi-options`, there are other tools to write and read EFI variables.

Reply:
Regarding bootloader spec, well grub implements superset of what we define. Hence I am not sure you want to mix and match bootloader tools. I'd say you either stick to grub or use systemd-boot.

Reply:
You are right. Especially, reading the current variable setting could be a problem without `bootctl` as that is stored in /run/systemd/efivars/SystemdOptions-UUID file and that is an implementation detail.

Hence, users can't set/read systemd options via EFI variable but we can get rid of gnu-efi dependency and don't ship bootctl. At least for RHEL/CentOS we are OK with the tradeoff. In general there is much bigger crowd interested in smaller system footprint than setting systemd options via EFI variables (which is only "auxiliary" as the main mechanism is still kernel cmdline).

Reply:
well, then add ENABLE_BOOTCTL, as an explicit knob. then people can select this.

Reply:
Honestly, 3 different knobs hence 8 possible combinations and out of those only 3 really make sense (full EFI, no EFI, just EFI in PID1) seems like overkill to me.

Reply:
On the other hand, I want the patch in so I'll do it if you insist. @keszybz should also weigh in with the opinion.